src/main/java/cz/actis/tutorial/testers/Tester4.java

```diff
@@ -12,6 +12,32 @@
      * Argumenty jsou double, double, double
      * Vraci se List<Point>, ktery bude obsahovat 0, 1 nebo 2 body.
      * Point vytvor pres new Point(x, y), kde x je souradnice na ose x, a y souradnice na ose y.
+     * <p>
+     * return
      */
-
+    public List<Point> solveQuadraticEquation(double a, double b, double c) {
+        double d = b * b - (4 * a * c);
+        List<Point> root = new ArrayList<>();
+        double y = 0;
+        if (d > 0) {
+            double x1 = (-b - Math.sqrt(d)) / (2 * a);
+            double x2 = (-b + Math.sqrt(d)) / (2 * a);
+            if (x1 < x2) {
+                Point root1 = new Point(x1, y);
```

Comment:
root1 a root2 nemusi byt v `if`u, ale pred nim.
A nepouzivej index u `root.add`, je to zbytecne.